web/html/src/components/input/FormMultiInput.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import * as React from 'react';
+
+/**
+ * Compute the list of items in the model based of fields named like `${prefix}${idx}_${name}`
+ */
+export function getOrderedItemsFromModel(model: Object, prefix: string): Array<string> {
+  return Object.keys(model)
+    .map(property => {
+      const result = property.match(new RegExp(`^${prefix}[0-9]+`));
+      if (result != null) {
+        return result[0];
+      }
+      return '';
+    })
+    .filter((property, index, array) => property !== '' && index === array.indexOf(property))
+    .sort((prop1, prop2) => {
+      const num1 = Number.parseInt(prop1.substring(prefix.length), 10);
+      const num2 = Number.parseInt(prop2.substring(prefix.length), 10);
+      return num1 - num2;
+    });
+}
```

Comment:
I wonder if it would make the component simpler if we make this method return an array of indices instead. A couple of advantages for example; in this method, you can map keys to indices, then they can be naturally sorted. And also, you can get rid of all the `substring` and `parseInt` calls in the render function as well.

Reply:
Done